src/item_location.cpp

```diff
@@ -31,7 +33,12 @@ class item_location::impl
         virtual void remove_item() = 0;
 
     protected:
-        item *what = nullptr;
+        impl( item_uid &&id ) : id( std::move( id ) ) {}
+
+        virtual item *target() const = 0;
+
+        mutable item *what = nullptr;
```

Comment:
This is the last member you should ever mark mutable, it's the core data for the class.